lib/node.js

```diff
@@ -192,18 +191,33 @@ Node.prototype._deleteRecursive = function (key, distinct, index) {
     return this;
   }
 
+  if (index === key.length) {
+    if (this.children[undefined]) {
+      this.children[undefined]._deleteRecursive(key, distinct, index + 1);
+    }
+    return this;
+  }
+
   const chr = key[index];
   if (!this.children[chr]) {
     // No such child exists, nothing to delete.
     return this;
   }
 
   // Recursively traverse the tree to find the node to delete.
-  this.children[chr] = this.children[chr]._deleteRecursive(key, distinct, index + 1);
+  this.children[chr] = this.children[chr]._deleteRecursive(
+    key,
+    distinct,
+    index + 1
+  );
 
   // If the child node has no values or children after deletion, remove it.
   if (!this.children[chr]) {
     delete this.children[chr];
+    const charIdx = this.values.indexOf(chr);
+    if (charIdx >= 0) {
```

Comment:
Non-trivial change here.

Reply:
Thanks for pointing out the specific changes to focus on. It's really helpful!